app/models/topic_tracking_state.rb

```diff
@@ -65,17 +76,32 @@ def self.publish_new(topic)
   def self.publish_latest(topic, staff_only = false)
     return unless topic.regular?
 
+    tags, tag_ids = nil
+    if SiteSetting.tagging_enabled
+      topic.tags.pluck(:id, :name).each do |id, name|
```

Comment:
Any reason not to use `transpose` here too?

Reply:
No reason, just missed it :)